plugins/binanceExchange_ws.go

```diff
@@ -180,6 +262,98 @@ func subcribeTicker(symbol string, state *mapEvents) (*stream, error) {
 
 }
 
+func subcribeUserStream(listenKey string, state *mapEvents) (*stream, error) {
+
+	userStreamLock := sync.Mutex{}
+
+	wsUserStreamExecutinReportHandler := func(message []byte) {
+
+		event := &eventExecutionReport{}
+		err := json.Unmarshal(message, event)
+
+		if err != nil {
+			log.Printf("Error unmarshal %s to eventExecutionReport\n", string(message))
+			return
+		}
+
+		userStreamLock.Lock()
+		defer userStreamLock.Unlock()
+
+		history, isHistory := state.Get(event.Symbol)
+
+		if !isHistory {
+			history.data = make(History, 0)
+			state.Set(event.Symbol, history)
+		}
+
+		now := time.Now()
+		history.createdAt = now
+
+		data, isOk := history.data.(History)
+
+		if !isOk {
+			log.Printf("Error conversion %v\n", ErrConversionHistory)
+			return
+		}
+
+		history.data = append(data, event)
+
+		lastCursor := event.TransactionTime
+
+		lastCursorData, isCursor := state.Get(LAST_CURSOR_KEY)
+
+		if isCursor {
+
+			cursor, isOk := lastCursorData.data.(int64)
+
+			if isOk {
+				if cursor > lastCursor {
+					lastCursor = cursor
+				}
+			} else {
+				log.Printf("Error converting cursor %v\n", ErrConversionCursor)
```

Comment:
maybe we should return an error here rather than just logging. we want the bot to crash if there is any error

Reply:
oke, I will return an error for conversion